API.md

```diff
@@ -676,12 +676,11 @@ curl http://localhost:<port>/irods-http-api/<version>/data-objects \
     -H 'Authorization: Bearer <token>' \
     --data-urlencode 'op=copy' \
     --data-urlencode 'src-lpath=<string>' \ # The absolute logical path of the data object to copy.
-    --data-urlencode 'dst-lpath=<string>' \ # The absolute logical path of the new data object.
-    --data-urlencode 'options=<string>' # skip_existing, overwrite_existing, update_existing, none. Defaults to none.
+    --data-urlencode 'dst-lpath=<string>' \ # The absolute logical path of a new or existing data object.
+    --data-urlencode 'resource=<string>' \ # The name of the root resource to copy to. Optional.
```

Comment:
Makes sense.

If we expose a source resource parameter, that means we can't use the filesystem lib for the copy operation and that's perfectly okay.

Do we have an issue in irods/irods capturing and/or explaining why `icp` doesn't expose the source resource option?

Reply:
Right. This PR only implements copy for data objects at the moment so we can ignore collections for now.

src/dst-resource seem like fine parameters to me.

Reply:
`src-resource` and `dst-resource` are now implemented.

This is possible because I scrapped use of the filesystem library. The implementation is much simpler and aligns with the goal of the project better.